sdk/spring/azure-spring-boot/src/main/java/com/azure/spring/autoconfigure/b2c/AADB2CAutoConfiguration.java

```diff
@@ -3,54 +3,38 @@
 package com.azure.spring.autoconfigure.b2c;
 
 import com.azure.spring.telemetry.TelemetrySender;
-import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnResource;
-import org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingClass;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Conditional;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
 import org.springframework.lang.NonNull;
-import org.springframework.security.oauth2.client.registration.ClientRegistration;
 import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
-import org.springframework.security.oauth2.core.AuthorizationGrantType;
-import org.springframework.security.oauth2.core.ClientAuthenticationMethod;
-import org.springframework.util.Assert;
 import org.springframework.util.ClassUtils;
 
 import javax.annotation.PostConstruct;
-import java.util.ArrayList;
 import java.util.HashMap;
-import java.util.List;
 import java.util.Map;
 
 import static com.azure.spring.telemetry.TelemetryData.SERVICE_NAME;
 import static com.azure.spring.telemetry.TelemetryData.TENANT_NAME;
 import static com.azure.spring.telemetry.TelemetryData.getClassPackageSimpleName;
 
 /**
- * {@link EnableAutoConfiguration Auto-configuration} for AAD B2C Authentication.
- * <p>
- * The configuration will not be activated if no {@literal azure.activedirectory.b2c.client-id,
- * client-secret and sign-in-user-flow} property provided.
- * </p>
+ * When the configuration matches the {@link AADB2COAuth2ClientConfiguration.AADB2CCondition.WebAppMode} condition,
+ * configure the necessary beans for AAD B2C authentication and authorization,
+ * and import {@link AADB2COAuth2ClientConfiguration} class for AAD B2C OAuth2 client support.
  */
 @Configuration
-@ConditionalOnWebApplication
-@ConditionalOnResource(resources = "classpath:aadb2c.enable.config")
-@ConditionalOnProperty(
-    prefix = AADB2CProperties.PREFIX,
-    value = {
-        "client-id",
-        "client-secret"
-    }
-)
+@Conditional({ AADB2COAuth2ClientConfiguration.AADB2CCondition.class })
+@ConditionalOnMissingClass({"org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken"})
```

Comment:
Is it possible for us to enable this AADB2CAutoConfiguration and AADB2CResourceServerAutoConfiguration at the same time?